src/app/pages/storage/modules/pool-manager/utils/generate-vdevs/grouped-disks.class.ts

```diff
@@ -10,30 +10,34 @@ import { getDiskTypeSizeMap } from 'app/pages/storage/modules/pool-manager/utils
 export class GroupedDisks {
   private diskMap: DiskTypeSizeMap;
 
-  constructor(
-    private disks: UnusedDisk[],
-  ) {
+  constructor(private disks: UnusedDisk[]) {
     this.diskMap = getDiskTypeSizeMap(disks);
   }
 
   findSuitableDisks(category: PoolManagerTopologyCategory): UnusedDisk[] {
-    const matchingType = this.diskMap[category.diskType];
     let matchingTypeAndSize: UnusedDisk[] = [];
+    const ssdDisks = this.diskMap.SSD[category.diskSize] || [];
+    const hddDisks = this.diskMap.HDD[category.diskSize] || [];
+
     if (category.treatDiskSizeAsMinimum) {
-      for (const diskSize of Object.keys(matchingType)) {
+      for (const diskSize of Object.keys(this.diskMap.HDD)) {
+        if (category.diskSize > Number(diskSize)) continue;
+        matchingTypeAndSize = matchingTypeAndSize.concat(this.diskMap.HDD[diskSize]);
+      }
+      for (const diskSize of Object.keys(this.diskMap.SSD)) {
```

Comment:
Can we make this prettier (i.e. unify two cases)?

Reply:
yes, it will be made as a part of refactoring, first of all I wanted to know that I'm moving in the right direction.